File: Sell.Ge FrontEnd/src/app/get/purchase.ts
export interface PurchaseItem {
  id: number;
  clothId?: number | null;
  shoeId?: number | null;
  itemName: string;
  picture: string;
  size?: string | null;
  unitPrice: number;
  quantity: number;
  lineTotal: number;
}

export interface Purchase {
  id: number;
  buyerId: number;
  buyerEmail: string;
  buyerFirstName: string;
  buyerLastName: string;
  buyerAddress?: string | null;
  buyerZipCode?: string | null;
  totalItems: number;
  totalPrice: number;
  purchasedAt: string;
  isDelivered: boolean;
  deliveredAt?: string | null;
  deliveredByAdminId?: number | null;
  items: PurchaseItem[];
}

File: Sell.Ge BackEnd/Sell.Ge/Models/Entities/Purchase/Purchase.cs
namespace Sell.Ge.Models.Entities.Purchase
{
    public class Purchase
    {
        public int Id { get; set; }

        public int BuyerId { get; set; }
        public string BuyerEmail { get; set; }
        public string BuyerFirstName { get; set; }
        public string BuyerLastName { get; set; }
        public string BuyerAddress { get; set; }
        public string BuyerZipCode { get; set; }

        public int TotalItems { get; set; }
        public decimal TotalPrice { get; set; }

        public DateTime PurchasedAt { get; set; } = DateTime.UtcNow;

        public bool IsDelivered { get; set; } = false;
        public DateTime? DeliveredAt { get; set; }
        public int? DeliveredByAdminId { get; set; }

        public List<PurchaseItem> Items { get; set; } = new List<PurchaseItem>();
    }
}
